Update Configuration Reference documentation (#2239)
withastro/astro: docs/src/pages/reference/configuration-reference.md

title: Configuration Reference
---

To configure Astro, add an `astro.config.mjs` file to the root of your project.

```js
export default /** @type {import('astro').AstroUserConfig} */ ({
  projectRoot: './',
  public: './public/',
  dist: './dist/',
  src: './src/',
  pages: './pages/',
  renderers: [

>     '@astrojs/renderer-svelte',

Reviewer: I think the renderers should be commented out, that way should they need them they merely need to uncomment them to turn them on,

Author: These represent the default values. I’ve added a note about this at the top of the object. <sup>[*](https://github.com/withastro/astro/pull/2239/files#diff-04d7d285adad1eb8d5e86bf573dd5a87657acd0ebfd9b0c30c77e9de40aa820bR10)</sup>

    '@astrojs/renderer-vue',
    '@astrojs/renderer-react',
    '@astrojs/renderer-preact',
  ],

> });

Reviewer: we should include a `vite:{ } ` option with a demo of how its written, perhaps applying a plugin,

```

All settings are optional.

The `projectRoot` option sets the working directory used by Astro. All other paths will be resolved from this. By default, this is the current working directory.

The `public` option sets the directory used to resolve public assets. By default, this is the `public` directory.

> The `dist` option sets the directory used to write the production build of the project.

Reviewer: ```suggestion
The `dist` option sets the directory used to output the final build of the project.
```

The `src` option sets the directory used to resolve source files, like `pages`. By default, this is the `src` directory.

> The `pages` option sets the directory used to resolve pages, relative to the `src` option. By default, this is the `pages` directory.

Reviewer: ```suggestion
The `pages` option sets the directory used to resolve pages, relative to the `src` option. By default, this is the `./src/pages` directory.
```

> The `renderers` option assigns the framework renderers to be used by Astro. By default, Astro provides renderers for Svelte, Vue, React, and Preact. To assign no renderers at all, provide an empty array (`[]`).

Reviewer: ```suggestion
The `renderers` opt-in option assigns the relevant framework renderers to be used by Astro. By default, Astro provides renderers for Svelte, Vue, React, and Preact. To assign no renderers at all, provide an empty array (`[]`).
```

Reviewer 2: Tiny comment to say that I think the phrase "By default, Astro provides..." is doing a lot of work here. 

I could see this being interpreted as "included in your project out of the box" (especially in conjunction with "default" which makes you think of the state you'll find when you first initialize a new project) vs what I think is more accurately "available to be added to your project." 

Would Fuzzy be happier with leaving the first sentence alone, but then something like:

"Astro *optionally* allows you to specify renderers for ... To assign no renderers at all, provide an empty array (`[]`) or comment out the field entirely." <-- assuming it's not commented out to begin with.

Carry on!

Author: I’ve rewritten all sections to avoid "By default" doing so much work.

The `buildOptions` option configures your site, including the site URL (`buildOptions.site`), whether it should generate a sitemap (`buildOptions.sitemap`), and whether pages should be written as `path.html` or `path/index.html` (`buildOptions.pageUrlFormat`).

The `devOptions` option configures your site for development, including the hostname (`devOptions.hostname`), the default port (`devOptions.port`), and whether urls should include a trailing slash (`devOptions.trailingSlash`).

> The `vite` option configures the internals of Vite. These options can be explored on [ViteJS.dev](https://vitejs.dev/config/).

Reviewer: should we elaborate a bit more on this?
I mean we should prehaps say something along the lines of  "All configurations that are applied in vite are applied to Astro, these allow for modification and fine adjustments over how vite is configured for the project"

Author: I’m not ready to add that, yet. That’s my own limitation. I’m actually not sure if it’s accurate that all Vite configurations are applied to Astro. Is it okay to punt on this for now?
